pkg/server/tunnel.go

```diff
@@ -72,6 +73,7 @@ func (t *Tunnel) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		Mode:      "http-connect",
 		HTTP:      conn,
 		connected: connected,
+		start:     time.Now(),
```

Comment:
nit: I'm sure the difference is negligible, but for consistency purposes can we arrange the dial flow in the same order between http connect and grpc? (Seems like in http-connect we do the backend selection and package sending after recording the start time, but in grpc those two steps are done before).

Reply:
Good point. I rearranged the order so that in both cases the latency includes the time of sending the dial_req.

This also fixed a memory leak in the tunnel scenario. Previously even if there is no backend available, a pending dial is added to server's pendingDialManager, but it will never be removed from the manager since the request will not even been sent out.